src/cometh/rental-protocol/rental-offer-created-handler.ts

```diff
@@ -0,0 +1,65 @@
+import { BigInt } from '@graphprotocol/graph-ts'
+import { RentalOffer } from '../../../generated/schema'
+import { COMETHSPACESHIP } from '../../utils/constants'
+import { RentalOfferCreated } from '../../../generated/ComethRentalProtocol/ComethRentalProtocol'
+import { loadNfts } from '../../utils/misc'
+import { log } from '@graphprotocol/graph-ts'
+import { SPACESHIP_ADDRESS } from '../../utils/addresses'
+/**
+ *    event RentalOfferCreated(
+ *       uint256 indexed nonce,
+ *       address indexed maker,
+ *       address taker,
+ *       NFT[] nfts,
+ *       address feeToken,
+ *       uint256 feeAmount,
+ *       uint256 deadline
+ *   );
+ */
+export function handleRentalOfferCreated(event: RentalOfferCreated): void {
+  // filter out nfts that are not spaceships
+  const filteredNfts = event.params.nfts.filter((nft) => nft.token.toHexString() == SPACESHIP_ADDRESS)
+
+  const filteredTokenIds = filteredNfts.map<BigInt>((nft) => nft.tokenId)
+
+  if (filteredTokenIds.length == 0) {
+    log.debug('[handleRentalOfferCreated] No spaceship nfts with valid address found in rental offer, tx: {}', [
+      event.transaction.hash.toHex(),
+    ])
+    return
+  }
+
+  const nfts = loadNfts(filteredTokenIds, COMETHSPACESHIP)
+
+  if (!nfts.length) {
```

Comment:
```suggestion
  if (nfts.length !== filteredTokenIds.length) {
```

Reply:
This is not possible since we can have mixed spaceships, bspaceships  and we want only to take the spaceships rentals. So we cannot skip offer creation if one of the nfts are not spaceship